### ISSUE TITLE
Update Pulls All Content

### ISSUE THREAD

**Issue by nebutron** [martinrotter/rssguard#103]
#### Brief description of the issue.

Using RSS Guard on Antergos and when using the Update All Items button it does a few different things. Sometimes it will grab everything the feed has, 1000s of items (old and new). Sometimes it will grab my YouTube folder and feeds within it (all of them, not new ones) but grab new items of other feeds outside of the folder.

Sometimes it works normally where it only grabs new items but it seems random.

#### How to reproduce the bug?

1. Open RSS Guard 3.4.0 on latest Antergos Cinnamon (not cloud accounts, local database)
2. Click the green Update All Items Button
3. Mark Everything As Read
4. Minimise to Tray
5. Click to Open, hit Update All Items Button Again
6. Watch something different happen each time.

#### What is the expected result?

Clicking the Update All Items button should grab any new posts from these feeds and nothing that is old or already read.

#### What actually happened?

It grabs all posts from the different feeds, old and previously read items. Sometimes it does this to all feeds, sometimes to only feeds within a folder.

**Comment by martinrotter** [comment]
Give me here exact feed links which do that.

RSS Guard can mark some feed's messages as new again, if it gets new content for the message.

**Comment by nebutron** [comment]
Well it seems to mostly be my YouTube RSS feeds so I will give one in example

https://www.youtube.com/feeds/videos.xml?channel_id=UCBJycsmduvYEL83R_U4JriQ

They are all doing it though. I suppose these videos are getting comments and causing the issue but my older RSS reader (RSSOwl) was not doing this.

I do now see this feed set to ATOM 1.0 instead of RSS 2.0, not sure if that matters?

**Comment by martinrotter** [comment]
OK, I will do some testing tomorrow. I see, the comments under each message COULD BE the real issue. RSS Guard checks if the message has new date/time. If it does, it overwrites the message in the database and sets it as unread. Perhaps, this does not make particular sense for YouTube feeds.

**Comment by martinrotter** [comment]
I am checking those feeds. It seems they contain up to 15 NEWEST entries (messages) even if the YouTube channel has more videos.

When any video gets new comment of anything changes, it is pushed into the feed and replaces some other older message in it, so that feed contains still max 15 messages. In RSS Guard, this will look like this:

- you get 15 messages,
- (now some video gets new comment or is updated or whatever)
- you get new X messages (the X messages which were updated or commented on).

But the actual contents of those messages are UNCHANGED. Therefore I made the change to take contents changes into account too. Now, the message will re-appear only if contents is changed too.
